packages/backend/src/errors/step.ts

```diff
@@ -1,9 +1,24 @@
-import { IStepError } from '@plumber/types'
+import { IJSONObject, IStepError } from '@plumber/types'
+
+import HttpError from './http'
 
 export default class StepError extends Error {
-  constructor(error: IStepError, options?: ErrorOptions) {
-    const computedMessage = JSON.stringify(error)
-    super(computedMessage, options)
+  constructor(
+    name: string,
+    solution: string,
+    position: number,
+    appName: string,
+    error?: HttpError,
+  ) {
+    const stepError: IStepError = {
+      name,
+      solution,
+      position,
+      appName,
+      details: error?.details as IJSONObject,
```

Comment:
nit: is there any way to avoid this typecast? Generally we try to avoid doing this unless absolutely necessary

Reply:
I think the `details` is a `Record<string, unknown>` in the `base.ts` file, which is a superset of the type `IJSONObject`, not sure how else to avoid this other than to just amend the type in the `base.ts` file instead

Reply:
OK. I think we leave any refactoring around this out for now.